src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -175,10 +191,36 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
         logger.info(`Leaving call session in room ${this.room.roomId}`);
         this.relativeExpiry = undefined;
         this.activeFoci = undefined;
+        this.encryptMedia = false;
         this.emit(MatrixRTCSessionEvent.JoinStateChanged, false);
         this.updateCallMembershipEvent();
     }
 
+    /**
+     * Re-sends the encryption key room event with a new key
+     */
+    private async updateEncryptionKeyEvent(): Promise<void> {
+        if (!this.isJoined()) return;
+        if (!this.encryptMedia) return;
+
+        logger.info(`MatrixRTCSession updating encryption key for room ${this.room.roomId}`);
+
+        const encryptionKey = getNewEncryptionKey();
+        await this.client.sendEvent(this.room.roomId, EventType.CallEncryptionPrefix, {
+            "m.encryption_key": encryptionKey,
+            "m.device_id": this.client.getDeviceId(),
+        } as EncryptionKeyEventContent);
+
+        const userId = this.client.getUserId();
+        const deviceId = this.client.getDeviceId();
+
+        if (!userId) throw new Error("No userId");
+        if (!deviceId) throw new Error("No deviceId");
+
+        this._encryptionKeys.set({ userId, deviceId }, encryptionKey);
```

Comment:
This is confusing me. Isn't `encryptionKeys` going to be for incoming media? Or do we also store our own in the same map but used for encrypting rather than decrypting? Personally I would store it in its own variable since it *is* a different thing.

Reply:
I more or less modeled this after LK which uses the same method to set the encryption key for the local user and decryption keys for remote ones. It does make sense to a degree since the cipher is symmetric, so they actually are almost the same thing

It really depends on perspective. I can change it, if you feel strongly about it

Reply:
Yeah, I guess it is similar to how livekit does it, so if it makes it easier there then I guess it makes sense. Otherwise it seems like it would be a map-lookup to get our own key rather than just a regular variable.